### PR TITLE
wiring CDC to QValue type system

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

File: flow/workflows/setup_flow.go [hunk 1]
```diff
@@ -207,7 +207,7 @@ func (s *SetupFlowExecution) fetchTableSchemaAndSetupNormalizedTables(
 
 		tableNameSchemaMapping[flowConnectionConfigs.TableNameMapping[srcTableName]] = srcTableSchema
 
-		s.logger.Info("setting up normalized table for table %s for peer flow - ", srcTableSchema, s.PeerFlowName)
+		s.logger.Info("setting up normalized table for table %s for peer flow - %s", srcTableName, s.PeerFlowName)
```
Comment (iskakaushik) [inline flow/workflows/setup_flow.go:210]: this has to be `Infof`

File: flow/model/qvalue/etime.go [hunk 2]
```diff
@@ -15,26 +15,26 @@ const (
 
 type ExtendedTime struct {
 	time.Time
-	NestedKind NestedKind
+	NestedKind ExtendedTimeKind
```
Comment (iskakaushik) [inline flow/model/qvalue/etime.go:18]: do we even need this ExtendedTimeKind, maybe this whole struct should be removed.

File: flow/connectors/snowflake/qvalue_convert.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package connsnowflake
+
+import (
+	"github.com/PeerDB-io/peer-flow/model/qvalue"
+)
+
+func getSnowflakeTypeForGenericColumnType(colType string) string {
```
Comment (iskakaushik) [inline flow/connectors/snowflake/qvalue_convert.go:7]: call this qValueTypeToSnowflakeType and same for bq

File: flow/workflows/setup_flow.go (hunk 1 above)
Comment (heavycrystal) [reply flow/workflows/setup_flow.go:210]: doesn't appear to be present in the Temporal `log.Logger` type

File: flow/connectors/bigquery/bigquery.go [hunk 4]
```diff
@@ -1003,40 +965,42 @@ func (m *MergeStmtGenerator) generateFlattenedCTE() string {
...
 		// Expecting interval to be in the format of {"Microseconds":2000000,"Days":0,"Months":0,"Valid":true}
 		// json.Marshal in SyncRecords for Postgres already does this - once new data-stores are added,
 		// this needs to be handled again
-		case model.ColumnTypeInterval:
-			castStmt = fmt.Sprintf("MAKE_INTERVAL(0,CAST(JSON_EXTRACT_SCALAR(_peerdb_data, '$.%s.Months') AS INT64),"+
-				"CAST(JSON_EXTRACT_SCALAR(_peerdb_data, '$.%s.Days') AS INT64),0,0,"+
-				"CAST(CAST(JSON_EXTRACT_SCALAR(_peerdb_data, '$.%s.Microseconds') AS INT64)/1000000 AS  INT64)) AS %s",
-				colName, colName, colName, colName)
-		case model.ColumnHexBit:
+		// TODO add interval types again
```
Comment (iskakaushik) [inline flow/connectors/bigquery/bigquery.go:988]: @heavycrystal  can you file an issue for this and time?

File: flow/connectors/bigquery/qvalue_convert.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	case qvalue.QValueKindJSON:
+		return bigquery.StringFieldType
+	// time related
+	case qvalue.QValueKindTimestamp, qvalue.QValueKindTimestampTZ:
+		return bigquery.TimestampFieldType
+	case qvalue.QValueKindDate:
+		return bigquery.DateFieldType
+	case qvalue.QValueKindTime, qvalue.QValueKindTimeTZ:
+		return bigquery.TimeFieldType
+	// TODO: handle INTERVAL types again
```
Comment (iskakaushik) [inline flow/connectors/bigquery/qvalue_convert.go:36]: link the issue in all the places in code as well, something like
`TODO (<issue_url>): description`

Comment (iskakaushik) [reply flow/connectors/bigquery/qvalue_convert.go:36]: we don't support `DATE` as well right?

File: flow/connectors/postgres/postgres_cdc_test.go [hunk 6]
```diff
@@ -603,36 +604,36 @@ func (suite *PostgresCDCTestSuite) TestAllTypesHappyFlow() {
 	suite.Equal(&protos.TableSchema{
 		TableIdentifier: allTypesHappyFlowSrcTableName,
 		Columns: map[string]string{
-			"id":  model.ColumnTypeInt64,
-			"c1":  model.ColumnTypeInt64,
+			"id":  string(qvalue.QValueKindInt64),
+			"c1":  string(qvalue.QValueKindInt64),
 			"c2":  model.ColumnHexBit,
```
Comment (iskakaushik) [inline flow/connectors/postgres/postgres_cdc_test.go:609]: why are these still `model.*` ?

File: flow/connectors/snowflake/snowflake.go [hunk 7]
```diff
@@ -768,19 +723,20 @@ func (c *SnowflakeConnector) generateAndExecuteMergeStatement(destinationTableId
...
-		case model.ColumnHexBit:
+		case qvalue.QValueKindBit:
 			// "c2": {"Bytes": "gA==", "Len": 1,"Valid": true}
 			flattenedCastsSQLArray = append(flattenedCastsSQLArray, fmt.Sprintf("BASE64_DECODE_BINARY(%s:%s:Bytes) "+
 				"AS %s,", toVariantColumnName, columnName, columnName))
-		case model.ColumnTypeTime:
-			flattenedCastsSQLArray = append(flattenedCastsSQLArray, fmt.Sprintf("TIME_FROM_PARTS(0,0,0,%s:%s:"+
-				"Microseconds*1000) "+
-				"AS %s,", toVariantColumnName, columnName, columnName))
+		// handle time types and interval types
```
Comment (iskakaushik) [inline flow/connectors/snowflake/snowflake.go:735]: link the issue here as well

File: flow/workflows/setup_flow.go (hunk 1 above)
Comment (iskakaushik) [reply flow/workflows/setup_flow.go:210]: If it's info does it support `%s` ?

File: flow/connectors/postgres/postgres_cdc_test.go (hunk 6 above)
Comment (heavycrystal) [reply flow/connectors/postgres/postgres_cdc_test.go:609]: there isn't any implementation for some types currently, I left them as placeholders so we can replace with new types once implemented

File: flow/model/qvalue/etime.go (hunk 2 above)
Comment (heavycrystal) [reply flow/model/qvalue/etime.go]: removed

File: flow/connectors/bigquery/bigquery.go (hunk 4 above)
Comment (heavycrystal) [reply flow/connectors/bigquery/bigquery.go:988]: filed as #189

File: flow/workflows/setup_flow.go (hunk 1 above)
Comment (heavycrystal) [reply flow/workflows/setup_flow.go:210]: doesn't support, changed to use fmt.Sprintf

File: flow/connectors/snowflake/snowflake.go (hunk 7 above)
Comment (heavycrystal) [reply flow/connectors/snowflake/snowflake.go]: done

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**